README.md

```diff
@@ -1,135 +1,287 @@
 
 ```js
-var google = require('googleapis');
-var GoogleAuth = require('google-auth-library').GoogleAuth;
+const GoogleAuth = require('google-auth-library').GoogleAuth;
+const auth = new GoogleAuth();
+
+/**
+ * Acquire a client, and make a request to an API that's enabled by default.
+ */
+async function main() {
+  try {
+    const adc = await getADC();
+    const url = `https://www.googleapis.com/dns/v1/projects/${adc.projectId}`;
+    const res = await adc.client.request({url});
+    console.log(res.data);
+  } catch (e) {
+    console.error('Error making request.');
+    console.error(e);
+  }
+}
 
-var authFactory = new GoogleAuth();
-var dns = google.dns('v1');
+/**
+ * Instead of specifying the type of client you'd like to use (JWT, OAuth2, etc)
+ * this library will automatically choose the right client based on the environment.
+ */
+async function getADC() {
+  // Acquire a client and the projectId based on the environment. This method looks
+  // for the GCLOUD_PROJECT and GOOGLE_APPLICATION_CREDENTIALS environment variables.
+  const res = await auth.getApplicationDefault();
+  let client = res.credential;
 
-authFactory.getApplicationDefault(function(err, authClient) {
-  if (err) {
-    console.log('Authentication failed because of ', err);
-    return;
+  // The createScopedRequired method returns true when running on GAE or a local developer
+  // machine. In that case, the desired scopes must be passed in manually. When the code is
+  // running in GCE or a Managed VM, the scopes are pulled from the GCE metadata server.
+  // See https://cloud.google.com/compute/docs/authentication for more information.
+  if (client.createScopedRequired && client.createScopedRequired()) {
+    // Scopes can be specified either as an array or as a single, space-delimited string.
+    const scopes = ['https://www.googleapis.com/auth/cloud-platform'];
+    client = client.createScoped(scopes);
   }
-  if (authClient.createScopedRequired && authClient.createScopedRequired()) {
-    var scopes = ['https://www.googleapis.com/auth/cloud-platform'];
-    authClient = authClient.createScoped(scopes);
+  return {
+    client: client,
+    projectId: res.projectId
   }
+}
+
+main();
+```
+
+#### OAuth2 client
+
+This client comes with an [OAuth2][oauth] client that allows you to retrieve an access token and refreshes the token and retry the request seamlessly if you also provide an `expiry_date` and the token is expired. The basics of Google's OAuth2 implementation is explained on [Google Authorization and Authentication documentation][authdocs].
+
+In the following examples, you may need a `CLIENT_ID`, `CLIENT_SECRET` and `REDIRECT_URL`. You can find these pieces of information by going to the [Developer Console][devconsole], clicking your project --> APIs & auth --> credentials.
```

Comment:
teeny nit: Higher up in the doc we use `>` arrows to show the navigation path. Here we use `-->`.

Reply:
fixed